### PR TITLE
Adds recommendation to install Prettier extension

### PR DESCRIPTION
**Pull request by jweill-aws**
## References

#11362 

## Code changes

Adds an extension recommendation for the Prettier extension

## User-facing changes

Affects the developer experience: Visual Studio Code users who do not have the Prettier extension will be asked in a non-modal dialog whether they want to add it.

## Backwards-incompatible changes

None

### CODE REVIEW

**Comment by jasongrout** [comment]
I'm +1 on this.

File: .gitignore [hunk 1]
```diff
@@ -121,6 +121,5 @@ junit.xml
 # ms IDE stuff
 *.code-workspace
 .history
-.vscode
```
Comment (jtpio) [inline .gitignore:124]: How will this removal affect VS Code users that might have a custom `settings.json`? For example to store `python.pythonPath` pointing to their virtual environment.

Looks like they might commit this file by mistake if it is not ignored anymore?

Comment (jweill-aws) [reply .gitignore:124]: Added `.vscode/settings.json` as a more specific exclusion.

**Comment by jtpio** [comment]
Folks can also setup a global `.gitignore` on their machine (for example in `~/.gitignore`) to avoid committing other files in the `.vscode` folder (or from other IDEs).

**Review by jtpio — APPROVED** [review]
Thanks!

**Pull request merged**